drivers/frequency/adf5902/adf5902.c

```diff
@@ -0,0 +1,1046 @@
+		return FAILURE;
+
+	dev->tx_ramp_clk = init_param->tx_ramp_clk;
+
+	if ((init_param->tx_data_invert != AD5902_TX_DATA_INV_DISABLE)
+	    && (init_param->tx_data_invert != AD5902_TX_DATA_INV_ENABLE))
+		return FAILURE;
+
+	dev->tx_data_invert = init_param->tx_data_invert;
+
+	if (init_param->clk2_div_no > (ADF5902_CLK_DIV_SEL_3 + 1))
+		return FAILURE;
+
+	dev->clk2_div_no = init_param->clk2_div_no;
+
+	for (i = 0; i < dev->clk2_div_no; i++)
+		if(init_param->clk2_div[i] > ADF5902_MAX_CLK_DIV_2)
+			return FAILURE;
+
+	dev->clk2_div = init_param->clk2_div;
+
+	if((init_param->clk_div_mode != ADF5902_CLK_DIV_OFF) &&
+	    (init_param->clk_div_mode != ADF5902_FREQ_MEASURE) &&
+	    (init_param->clk_div_mode != ADF5902_RAMP_DIV))
+		return FAILURE;
+
+	dev->clk_div_mode = init_param->clk_div_mode;
+
+	if((init_param->le_sel != ADF5902_LE_FROM_PIN) &&
+	    (init_param->le_sel != ADF5902_LE_SYNC_REFIN))
+		return FAILURE;
+
+	dev->le_sel = init_param->le_sel;
+
+	if (init_param->cp_current > ADF5902_CP_CURRENT_4MA48)
+		return FAILURE;
+
+	dev->cp_current = init_param->cp_current;
+
+	if ((init_param->cp_tristate_en != ADF5902_CP_TRISTATE_DISABLE)
+	    && (init_param->cp_tristate_en != ADF5902_CP_TRISTATE_ENABLE))
+		return FAILURE;
+
+	dev->cp_tristate_en = init_param->cp_tristate_en;
+
+	if(init_param->ramp_status != ADF5902_TEST_BUS_NONE &&
+	    init_param->ramp_status != ADF5902_RAMP_COMPL_TO_MUXOUT &&
+	    init_param->ramp_status != ADF5902_RAMP_DOWN_TO_MUXOUT &&
+	    init_param->ramp_status != ADF5902_TEMP_SENS_TO_ATEST &&
+	    init_param->ramp_status != ADF5902_TEMP_SENS_TO_ADC)
+		return FAILURE;
+
+	dev->ramp_status = init_param->ramp_status;
+
+	if(init_param->ramp_mode != ADF5902_CONT_SAWTOOTH &&
+	    init_param->ramp_mode != ADF5902_SAWTOOTH_BURST &&
+	    init_param->ramp_mode != ADF5902_CONTINUOUS_TRIANGULAR &&
+	    init_param->ramp_mode != ADF5902_SINGLE_RAMP_BURST)
+		return FAILURE;
+
+	dev->ramp_mode = init_param->ramp_mode;
+
+	/* Set device specific parameters for obtaining the VCO frequency */
+	ret = adf5902_vco_freq_param(dev);
+	if (ret != SUCCESS)
+		return ret;
+
+	/* Set frequency calibration divider value */
+	dev->freq_cal_div = (uint16_t)(dev->f_pfd / ADF5902_FREQ_CAL_DIV_100KHZ);
+
+	/* f_PFD / 100kHz <= freq_cal_div */
+	if(dev->f_pfd % ADF5902_FREQ_CAL_DIV_100KHZ)
+		dev->freq_cal_div += 1;
+
+	if(dev->freq_cal_div > ADF5902_MAX_FREQ_CAL_DIV)
+		return FAILURE;
+
+	/* Set clock divider value */
+	dev->clk1_div = dev->f_pfd / ADF5902_CLK1_DIV_25KHZ;
+
+	/* f_PFD/25kHz <= clk1_div */
+	if (dev->f_pfd % ADF5902_CLK1_DIV_25KHZ)
+		dev->clk1_div += 1;
+
+	if (dev->clk1_div > ADF5902_MAX_CLK_DIVIDER)
+		return FAILURE;
+
+	/* Set ADC clock divider value for 1MHz ADC clock */
+	dev->adc_clk_div = dev->f_pfd / ADF5902_ADC_CLK_DIV_1MHZ;
+
+	if (dev->adc_clk_div > ADF5902_ADC_MAX_CLK_DIVIDER)
+		return FAILURE;
+
+	/* Master Reset */
+	ret = adf5902_write(dev, ADF5902_REG7 | ADF5902_REG7_RESERVED |
```

Comment:
What about adding something similar to: https://github.com/analogdevicesinc/linux/blob/master/drivers/iio/regmap/iio-regmap.c
In order to initialize the register from an array of commands?
(Not in this PR)